test/jit/test_with.py

```diff
@@ -548,3 +548,30 @@ def test_exit_incorrect_types(x, c):
             self.checkScript(
                 test_exit_incorrect_types, (test_tensor, ExitIncorrectTypes())
             )
+
+    def test_with_no_grad(self):
+        """
+        Check that torch.no_grad() works.
+        """
+        @torch.jit.script
+        class NoGrad(object):
```

Comment:
I put this class here temporarily, but I'd like to move it to a place where all users can access it and have `torch.no_grad` resolve to it somehow.